src/nft/hooks/useMintDetails.ts

```diff
@@ -0,0 +1,31 @@
+import { type UseQueryResult, useQuery } from '@tanstack/react-query';
+import { getMintDetails } from '../../api/getMintDetails';
+import type { GetMintDetailsParams, MintDetails } from '../../api/types';
+import { isNFTError } from '../utils/isNFTError';
+
+export function useMintDetails({
+  contractAddress,
+  takerAddress,
+  tokenId,
+}: GetMintDetailsParams): UseQueryResult<MintDetails> {
+  const actionKey = `useMintDetails-${contractAddress}-${takerAddress}-${tokenId}`;
+  return useQuery({
+    queryKey: ['useMintDetails', actionKey],
+    queryFn: async () => {
+      const mintDetails = await getMintDetails({
+        contractAddress,
+        takerAddress,
+        tokenId,
+      });
+
+      if (isNFTError(mintDetails)) {
+        throw mintDetails;
+      }
+
+      return mintDetails;
+    },
+    enabled: true,
```

Comment:
i think the default is true here so you might not need this. is there a case where we want to disable ?